api/v1alpha1/pipelinegate_types.go

```diff
@@ -99,6 +103,74 @@ func (pgs *PipelineGateStatus) HasJobRef() bool {
 	return !(pgs.JobRef == nil || *pgs.JobRef == console.NamespacedName{})
 }
 
-func (pgs *PipelineGateStatus) HasNotReported() bool {
-	return pgs.LastReported == nil || (pgs.LastReportedAt != nil && pgs.LastReportedAt.Before(&pgs.LastSyncedAt))
+//func (pgs *PipelineGateStatus) HasNotReported() bool {
+//	return pgs.LastReported == nil || (pgs.LastReportedAt != nil && pgs.LastReportedAt.Before(&pgs.LastSyncedAt))
+//}
+
+func (pgs *PipelineGateStatus) GetConsoleGateState() (*console.GateState, error) {
+	if pgs.State == nil {
+		return nil, fmt.Errorf("gate state is not initialized")
+	}
+	state := console.GateState(*pgs.State)
+	return &state, nil
+}
+
+func (pgs *PipelineGateStatus) GetSHA() string {
+	if !pgs.HasSHA() {
+		return ""
+	}
+	return *pgs.SHA
+}
+
+func (pgs *PipelineGateStatus) HasSHA() bool {
+	return pgs.SHA != nil && len(*pgs.SHA) > 0
+}
+
+func (pgs *PipelineGateStatus) IsSHAEqual(sha string) bool {
+	if !pgs.HasSHA() {
+		return false
+	}
+	return pgs.GetSHA() == sha
+}
+
+func (pgs *PipelineGateStatus) SetState(state console.GateState) *PipelineGateStatus {
+	gateState := GateState(state)
+	pgs.State = &gateState
+	return pgs
+}
+
+func (pgs *PipelineGateStatus) SetJobRef(name string, namespace string) *PipelineGateStatus {
+	nsn := console.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+	pgs.JobRef = &nsn
+	return pgs
+}
+
+func (pgs *PipelineGateStatus) GateUpdateAttributes() (*console.GateUpdateAttributes, error) {
+	state, err := pgs.GetConsoleGateState()
+	if err != nil {
+		return nil, err
+	}
+	updateAttributes := console.GateUpdateAttributes{State: state, Status: &console.GateStatusAttributes{JobRef: pgs.JobRef}}
+	return &updateAttributes, nil
+}
+
+func (pgs *PipelineGateStatus) SetSHA(sha string) *PipelineGateStatus {
+	pgs.SHA = &sha
+	return pgs
+}
+
+func (pg *PipelineGate) CreateNewJobName() string {
+	jobName := fmt.Sprintf("%s-%s", pg.Name, uuid.New().String()[:8])
```

Comment:
maybe it would be better to use this:
```
import  "k8s.io/apimachinery/pkg/util/rand"


jobName := fmt.Sprintf("%s-%s", pg.Name, rand.String(8))
```